daemon/api_sideload_n_try.go

```diff
@@ -107,6 +112,16 @@ func (f *Form) GetSnapFiles() ([]*uploadedContainer, *apiError) {
 		if len(f.Values["name"]) > 0 {
 			uploaded.instanceName = f.Values["name"][0]
 		}
+
+		if len(f.Values["component-ref"]) > 0 {
```

Comment:
Does this require a spec change? Also, in the API we tend to avoid the snap+comp syntax so maybe we should have 2 fields, snap-name and component-name or similar. Maybe having it separate would help to do things like sideloading snap instances in the future.

Reply:
Unsure, this isn't really talked about in the spec. This change is something that @pedronis and I talked about during the sprint. Splitting it into two seems fine for me. For the planned CLI changes, do you think that should still just be one param? (--component-ref?)

Reply:
I changed this to use the existing "name" form field and a new "component-name" field in 180bc2c28718b82a39c8f3177a2b60866bbc8880. Let me know what you think.

Reply:
Looks good, thanks for the change. For the cli note that we already have a `--name` for instances, so probably we still want `--component-name` there too, maybe @pedronis has a different opinion though. We could support both ways too.